book/src/platformer/03_importing_tiled_levels.md

```diff
@@ -0,0 +1,325 @@
+            // otherwise, use the blank tile
+            None => quote! { TileSetting::BLANK },
+        });
+
+        // this creates the bit-array from the booleans on the tiles.
+        let collision_map = self.tiles.chunks(8).map(|x| {
+            x.iter()
+                .map(|x| x.colliding as u8)
+                .fold(0u8, |a, b| (a >> 1) | (b << 7))
+        });
+
+        let winning_map = self.tiles.chunks(8).map(|x| {
+            x.iter()
+                .map(|x| x.win as u8)
+                .fold(0u8, |a, b| (a >> 1) | (b << 7))
+        });
+
+        let (player_x, player_y) = self.player_start;
+        let (width, height) = self.size;
+
+        // see how easy it is to define Rust code! See the quote documentation for more details.
+        quote! {
+            Level {
+                // just puts the width in
+                width: #width,
+                height: #height,
+                // this includes every element of the iterator separated by commas
+                background: &[#(#background_tiles),*],
+                collision_map: &[#(#collision_map),*],
+                winning_map: &[#(#winning_map),*],
+                player_start: (#player_x, #player_y),
+            }
+        }.to_tokens(tokens)
+    }
+}
+```
+
+# Driving the level output
+
+In our `build.rs` file, we need to include a main function to be run by `cargo`.
+
+```rust
+// the levels we should load
+static LEVELS: &[&str] = &["level_01.tmx"];
+
+fn main() -> Result<(), Box<dyn Error>> {
+    // the file we output to which is in the `OUT_DIR` directory. This is a directory
+    // provided by cargo designed to be used by the build script to include output into
+    let mut file =
+        std::fs::File::create(format!("{}/levels.rs", std::env::var("OUT_DIR").unwrap()))?;
+
+    // make and write each level to the output
+    for (number, level) in LEVELS.iter().enumerate() {
+        let ident = quote::format_ident!("LEVEL_{}", number);
+        let level = import_level(&format!("tiled/{level}"))?;
+        let content = quote! {
+            static #ident: Level = #level;
+        };
+        writeln!(file, "{content}")?;
+    }
+
+    // define an array of all the levels to be used by the game, therefore make the array `pub`.
+    let levels = (0..LEVELS.len()).map(|x| quote::format_ident!("LEVEL_{}", x));
+    writeln!(
+        file,
+        "{}",
+        quote! {
+            pub static LEVELS: &[&Level] = &[#(&#levels),*];
+        }
+    )?;
+
+    Ok(())
+}
+```
+
+# Summary
+
+Here we've made a level import system for our game.
+We've shown how you can use the basic Tiled features along with a more advanced one (custom properties).
+How you use tiled is highly individualised and as your projects grow so too will your use of tiled and the various features you use and have to write support for in your import system.
+To make levels quickly and efficiently, you will want to individualise it and make it work for you.
```

Comment:
```suggestion
How you use tiled is very specific and as your projects grow so too will your use of tiled and its various features. 
To make levels quickly and efficiently, you will want to specialise it and make it work for you.
```

Reply:
I'd like to convey that your use of tiled and your import system are linked. Our use of tiled in Khiera and DPL are completely different and have very different tooling.